### PR TITLE
move premium check to middleware

### LINKED ISSUE #9630: [FEATURE] Move premium user check to middleware
### Is this a unique feature?

- [X] I have checked "open" AND "closed" issues and this is not a duplicate

### Is your feature request related to a problem/unavailable functionality? Please describe.

For /pages/account/statistics we are using "getServerSideProps" redirection instead of using middleware.js.  The middleware ends code duplication and reduces possibility of introducing errors.

### Proposed Solution

I'm proposing we consolidate authentication checking using only middleware.js. 

### Screenshots

image

### Do you want to work on this issue?

Yes

### If you want to work on this issue...

I would add in middleware.js a code that would look at current user route and if it is /account/statistics and user account type is not premium, it would redirect him

### PR DESCRIPTION
**Pull request by dmitrykulakovfrontend**
## Fixes Issue
Closes #9630

## Changes proposed
Update JWT, refactor middleware.js, remove premium checks in getServerSideProps and move them to middleware.js

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

### CODE REVIEW

File: middleware.js [hunk 1]
```diff
@@ -14,43 +14,51 @@ export const config = {
...
   const reqPathName = req.nextUrl.pathname;
 
-  // if not in sessionRequired or adminRequired, skip
-  if (
-    !sessionRequired
-      .concat(adminRequired)
-      .some((path) => reqPathName.startsWith(path))
-  ) {
-    return NextResponse.next();
-  }
```
Comment (eddiejaoude) [inline middleware.js:29]: I think this is still required?

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you 👍 

Looks good, I left an inline question. 

Please can you also resolve conflicts

**Comment by eddiejaoude** [comment]
Thank you for making the changes. Could you check your Prettier settings, it looks like extra lines are being reformatted, which makes it more difficult to review "actual" changes

**Comment by dmitrykulakovfrontend** [comment]
I'm apologizing for the issue with prettier, it should be fixed now

**Comment by eddiejaoude** [comment]
> I'm apologizing for the issue with prettier, it should be fixed now

thank you 👍  sorry for my slow reply, I will be looking at this PR next

**Comment by eddiejaoude** [comment]
I can fix the conflicts, as it has conflicted with my changes

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍 

note: I resolved the conflicts

**Pull request merged**